src/liballoc/boxed.rs

```diff
@@ -63,6 +63,27 @@
 //! T` obtained from `Box::<T>::into_raw` may be deallocated using the
 //! [`Global`] allocator with `Layout::for_value(&*value)`.
 //!
+//! `Box<T>` has the same ABI as `&mut T`. In particular, when `T: Sized`,
+//! this allows using `Box<T>` in FFI:
```

Comment:
Cc @gnzlbg @rkruppe regarding our recent discussion about `noalias` leaking from declarations.

Note however that this is not a new thing we are adding here, the docs for reference types already state

> Option<&T> has the same memory representation as a nullable but aligned pointer, and can be passed across FFI boundaries as such.

So my inclination is to accept this wording, and figure out separately how we can document the side-effects this might have with xLTO?

Reply:
Agreed.

Reply:
What do you think would be the right issue tracker for that? Given that there are still open questions, I am leaning towards UCG.

Reply:
I don't have a strong opinion on where, I think a Rust issue could be suitable as well (especially considering the relation to #46188). I just want there to be an issue for it somewhere so we don't forget.

Reply:
I've opened an issue in the UCGs: https://github.com/rust-lang/unsafe-code-guidelines/issues/198